### PR TITLE
Release/flocker 0.3.3dev3

### PR DESCRIPTION
**Pull request by exarkun**
Fixes FLOC-1083

(Do not merge.  Merge will happen at a specified point in the release process.)

### CODE REVIEW

**Comment by adamtheturtle** [comment]
Note as follow-up work that the "Pre-tag Review Process" says to "Accept or reject the release issue depending on whether everything has worked.", referencing the acceptance tests. What I'm actually doing first is reading through the "Preparing For a Release" section and checking that the "Files changed" match up with what should be changed.

**Comment by adamtheturtle** [comment]
Similarly "Ensure all the tests pass on BuildBot:" has seemingly been skipped in favour of "Ensure all the tests pass on BuildBot, apart from those we know are failing and that is OK". Not quite as concise.

File: LICENSE [hunk 1]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright 2014 ClusterHQ
```
Comment (adamtheturtle) [inline LICENSE]: The documentation at this point is weird - it says:

```
Ensure copyright dates in LICENSE are up-to-date:

XXX: Process to be decided. See https://github.com/ClusterHQ/flocker/issues/525
```

However that is a related issue closed as "Done". Either create an issue to be linked to at this part of the docs or update them to say something like "If no releases have been finished earlier in this calendar year, add this year to the LICENSE file". I'll work on the assumption that what you have done here is right.

**Comment by adamtheturtle** [comment]
The changes look fine and I will do the acceptance tests now.

**Comment by adamtheturtle** [comment]
> Similarly "Ensure all the tests pass on BuildBot:" has seemingly been skipped in favour of "Ensure all the tests pass on BuildBot, apart from those we know are failing and that is OK". Not quite as concise.

Perhaps when this is the case it is best to say which tests / builders are failing and that you know this is OK.

**Comment by adamtheturtle** [comment]
The Pre-tag Review Process says

> The URL can be found by examining the “upload-base-box” step of the flocker-vagrant-tutorial-box builder. The URL will look like http://build.clusterhq.com/results/vagrant/<RELEASE_BRANCH>/flocker-tutorial.json.

This is incorrect. There is now a "tutorial box" link in the "write-base-box-metadata" step.

**Comment by adamtheturtle** [comment]
The Pre-tag Review Process says to run commands including:

```
git checkout -b *release branch*
mkvirtualenv flocker-release-${VERSION}
```

However, VERSION has not been set at this point and if it has then "*release branch*" can be determined ("release/flocker-${VERSION}").

**Comment by adamtheturtle** [comment]
The above has "git checkout -b" which is certainly wrong, you don't want to be creating a new branch.

**Comment by adamtheturtle** [comment]
The acceptance tests fail. When I run flocker-deploy I get the following error:

```
(flocker-release-flocker-0.3.3dev3)~/Documents/ClusterHQ/flocker-0.3.3dev3/flocker Adam$ flocker-deploy 1
Traceback (most recent call last):
  File "/Users/Adam/.virtualenvs/flocker-release-flocker-0.3.3dev3/bin/flocker-deploy", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/Users/Adam/.virtualenvs/flocker-release-flocker-0.3.3dev3/lib/python2.7/site-packages/pkg_resources.py", line 2829, in <module>
    working_set = WorkingSet._build_master()
  File "/Users/Adam/.virtualenvs/flocker-release-flocker-0.3.3dev3/lib/python2.7/site-packages/pkg_resources.py", line 449, in _build_master
    ws.require(__requires__)
  File "/Users/Adam/.virtualenvs/flocker-release-flocker-0.3.3dev3/lib/python2.7/site-packages/pkg_resources.py", line 742, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/Users/Adam/.virtualenvs/flocker-release-flocker-0.3.3dev3/lib/python2.7/site-packages/pkg_resources.py", line 639, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: backports.ssl-match-hostname
```
If I `pip install backports.ssl_match_hostname` then I can run `$ flocker-deploy` and get the expected flocker error message. The acceptance tests still fail with errors such as "Error connecting to cluster node". See https://gist.github.com/adamtheturtle/50f8c4faacdc7cec1394 for details.

Also, do what you think is best with my above comments.

I will look further into the acceptance test failures but it looks at least like backports.ssl_match_hostname should be added to the requirements.

File: LICENSE (hunk 1 above)
Comment (exarkun) [inline LICENSE]: #1036

**Comment by exarkun** [comment]
> Perhaps when this is the case it is best to say which tests / builders are failing and that you know this is OK.

Seems reasonable.  I'm going to look at making all of the builders actually be green again shortly.

**Comment by adamtheturtle** [comment]
Ok I won't look further into the IgnoreUnknown issue and I will leave this to you.

**Comment by exarkun** [comment]
> The Pre-tag Review Process says

> > The URL can be found by examining the “upload-base-box” step of the flocker-vagrant-tutorial-box builder. The URL will look like http://build.clusterhq.com/results/vagrant/.../flocker-tutorial.json.
> This is incorrect. There is now a "tutorial box" link in the "write-base-box-metadata" step.

There is also a "tutorial box" link there, yes - but I don't think the existing docs are incorrect.  I managed to construct the correct URL by inserting the release branch name into the given URL template and found the json file.

**Comment by adamtheturtle** [comment]
> There is also a "tutorial box" link there, yes - but I don't think the existing docs are incorrect.

Did you "examine" the “upload-base-box” step of the flocker-vagrant-tutorial-box builder? I didn't find the relevant URL there. If the URL will always fit the template and $VERSION needs to be set for a later step, perhaps the best thing is to have `vagrant box add http://build.clusterhq.com/results/vagrant/flocker-release-${VERSION}/flocker-tutorial.json` as the step, so you don't even need to go to the Buildbot site.

**Comment by exarkun** [comment]
> Did you "examine" the “upload-base-box” step of the flocker-vagrant-tutorial-box builder? 

I didn't.  The URL template is giving the pre-release review instructions.

> If the URL will always fit the template and $VERSION needs...

Good call.  I'll make this doc change for FLOC-1177.

**Comment by exarkun** [comment]
Further progress on this blocked on being unable to `vagrant up` the dev box - `default: Warning: Connection timeout. Retrying...` indefinitely.

**Comment by adamtheturtle** [comment]
I needed the acceptance tests to pass for another issue and found that commenting both occurrences of ``b"-o", b"IgnoreUnknown=GSSAPIAuthentication",`` seemed to work for the one test I'm working with.

**Comment by adamtheturtle** [comment]
The issue is marked as "Won't do" so I'm closing this.

**Pull request closed without merging**